lms/djangoapps/instructor/views/api.py

```diff
@@ -1277,66 +1277,77 @@ def post(self, request, course_id):
-        * problem_types_filter: Optional. A comma-separated list of block types
-          to include in the repot. If set, only blocks of the specified types will
-          be included in the report.
+            **POST Parameters**
 
-        To get data on all the poll and survey blocks in a course, you could
-        POST the usage key of the course for `problem_location`, and
-        "poll, survey" as the value for `problem_types_filter`.
+            A POST request can include the following parameters:
 
+            * problem_location: A comma-separated list of usage keys for the blocks
+              to include in the report. If the location is a block that contains
+              other blocks, (such as the course, section, subsection, or unit blocks)
+              then all blocks under that block will be included in the report.
+            * problem_types_filter: Optional. A comma-separated list of block types
+              to include in the repot. If set, only blocks of the specified types will
+              be included in the report.
 
-    **Example Response:**
-    If initiation is successful (or generation task is already running):
-    ```json
-    {
-        "status": "The problem responses report is being created. ...",
-        "task_id": "4e49522f-31d9-431a-9cff-dd2a2bf4c85a"
-    }
-    ```
-
-    Responds with BadRequest if any of the provided problem locations are faulty.
-    """
-    # A comma-separated list of problem locations
-    # The name of the POST parameter is `problem_location` (not pluralised) in
-    # order to preserve backwards compatibility with existing third-party
-    # scripts.
-    problem_locations = request.POST.get('problem_location', '').split(',')
-    # A comma-separated list of block types
-    problem_types_filter = request.POST.get('problem_types_filter')
-    return _get_problem_responses(
-        request,
-        course_id=course_id,
-        problem_locations=problem_locations,
-        problem_types_filter=problem_types_filter,
-    )
+            To get data on all the poll and survey blocks in a course, you could
+            POST the usage key of the course for `problem_location`, and
+            "poll, survey" as the value for `problem_types_filter`.
+
+
+        **Example Response:**
+        If initiation is successful (or generation task is already running):
+        ```json
+        {
+            "status": "The problem responses report is being created. ...",
+            "task_id": "4e49522f-31d9-431a-9cff-dd2a2bf4c85a"
+        }
+        ```
+
+        Responds with BadRequest if any of the provided problem locations are faulty.
+        """
+        # A comma-separated list of problem locations
+        # The name of the POST parameter is `problem_location` (not pluralised) in
+        # order to preserve backwards compatibility with existing third-party
+        # scripts.
+
+        problem_locations = request.POST.get('problem_location', '').split(',')
```

Comment:
This API has been converted to use DRF's APIView for consistency.
No changes have been made to the existing core logic, as doing so was causing test failures and I don't have full domain knowledge.

This change is limited to structural migration only.

A similar DRF-based API already exists in this [file](https://github.com/openedx/edx-platform/blob/master/lms/djangoapps/instructor/views/api.py#L1266).